lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_upload_pipeline_action.rb

```diff
@@ -0,0 +1,70 @@
+module Fastlane
+  module Actions
+    class BuildkiteUploadPipelineAction < Action
+      def self.run(params)
+        pipeline_env_file = params[:pipeline_env_file]
+        pipeline_file = params[:pipeline_file]
+        branch = params[:branch]
+        commit = params[:commit]
+
+        UI.user_error!("Pipeline file not found: #{pipeline_file}") unless File.exist?(pipeline_file)
+
+        UI.message "Uploading pipeline #{pipeline_file} on branch #{branch}, commit #{commit}"
+
+        ENV['BUILDKITE_BRANCH'] = branch
+        ENV['BUILDKITE_COMMIT'] = commit
+
+        if pipeline_env_file && File.exist?(pipeline_env_file)
+          UI.message("Sourcing environment file: #{pipeline_env_file}")
+
+          sh(". #{pipeline_env_file} && buildkite-agent pipeline upload #{pipeline_file}")
```

Comment:
I could't find a good way of sourcing the `pipeline_env_file`, making it available to the `buildkite-agent` call, while also keeping the split arguments.
Since the parameters in this string are files that are validated, I think there's less room for potential issues so I just kept it like this for now.

Reply:
> I don't see what you mean by those being validated?

I meant `validated` by the checks to make sure both files exist, so it makes a bit harder to inject commands in it.

Reply:
> I assume you tested having two separate calls of sh, but that didn't work probably because each call to sh spawns its own shell subprocess so they don't share their environment/context…

🎯 exactly!

For completeness, also mentioning here that I've added the `.shellescape` and did a few other `sh()` / env updates on 07661f76ab61bc32a665e8837c27c68a2a02c5f3.